applications/CompressiblePotentialFlowApplication/custom_processes/apply_far_field_process.cpp

```diff
@@ -0,0 +1,129 @@
+//    |  /           |
+//    ' /   __| _` | __|  _ \   __|
+//    . \  |   (   | |   (   |\__ `
+//   _|\_\_|  \__,_|\__|\___/ ____/
+//                   Multi-Physics
+//
+//  License:         BSD License
+//                   Kratos default license: kratos/license.txt
+//
+//  Main authors:    Iñigo Lopez and Marc Núñez
+//
+
+// Project includes
+#include "apply_far_field_process.h"
+
+namespace Kratos {
+
+// Constructor for ApplyFarFieldProcess Process
+ApplyFarFieldProcess::ApplyFarFieldProcess(ModelPart& rModelPart, const double rReferencePotential)
+    : Process(), mrModelPart(rModelPart), mReferencePotential(rReferencePotential)
+{
+    mFreeStreamVelocity = mrModelPart.GetProcessInfo()[FREE_STREAM_VELOCITY];
+}
+
+void ApplyFarFieldProcess::Execute()
+{
+    FindFarthestUpstreamBoundaryNode();
+    AssignFarFieldBoundaryConditions();
+}
+
+void ApplyFarFieldProcess::FindFarthestUpstreamBoundaryNode()
+{
+    // Declaring omp variables, generating vectors of size = num_threads
+    std::size_t num_threads = omp_get_max_threads();
```

Comment:
That should be inside a `#pragma omp parallel` region or it will return 1 all the time. (If its for this OmpUtils its ok I think)

Reply:
I checked this with KRATOS_WATCH() and was getting the correct number of threads according to the environmental variable `export OMP_NUM_THREADS`. At first I was using `omp_get_num_threads()` and I was getting 1 always as you say. 

Shall i use `omp_get_num_threads()` instead of `omp_get_max_threads()`?

Reply:
Its better yes. Not the case but you could have:

```C++
int num_threads = omp_get_max_threads();

#pragma omp parallel for num_threads(13) // being 13 != num_threads
for(...) {
}
```

And then the code will crash. Similar scenario with nested parallelism or dynamic load balancing. Mostly for corner cases but is safer in to use `omp_get_num_threads()`. Also some implementations may not return what you expect ( I am thinking on komp)

Reply:
Ok thanks, I will use get_num_threads then!

Reply:
This will not work if Kratos is compiled without omp support
I will write more later